### DISCUSSION TITLE
Open Source the code (APK)

### DISCUSSION THREAD

**Discussion by nkonidaris** [seaky/nspanel_pro_tools_apk#146] [Q&A]
Good day, @seaky 

Wanted to once again thank you for your hard work on making this app a reality! It really has given the NSPanel Pro a place in my home and works wonders.

I would like to ask if there is any plan to make the APK source files open source for review, assistance (pull-requests), and much more! Let me know your thoughts and look forward to a response!

**Comment by bobzer** [comment]
Hi, 
From my understanding, some part of the code can't be open source because of some proprietary shenanigans.
The things I'm wondering is how much it could be segregate and abstract. I know it would be some work, maybe not just some but a bif refactoring. Anyway I agree with nkonidaris it would be great to open source this great tool, and we could help you more. 
@seaky Do you thinks it's possible to have a dependency apk that handle and abstract the proprietary things and open source the rest ?
Anyway, this is a great project, thank you for all your work

**Comment by seaky** [comment]
I consider how to split source code to patented and open but not as easy as it looks, initially I wrote it to myself so there is lot of entanglement between open and proprietary.

**Reply by ivdimitro** [reply]
Just out of curiosity - where is the propreatory coming from? Is it a firmware of some controller or something else?

**Reply by seaky** [reply]
I reverse engineered an api which is integrated into AOSP and provided by the manufacturer and some ewelink related functions.
I'm not aware of that if you extend an open source OS like AOSP with your own code as a manufacturer is it proprietary or should be open source by default!
Btw the sonoff firmware's source code is not open, I assume that just the kernel should be open because it is on github! Or there is some cheat;)
